x/staking/types/delegation.go

```diff
@@ -231,6 +213,16 @@ func (e RedelegationEntry) OnHold() bool {
 	return e.UnbondingOnHoldRefCount > 0
 }
 
+// Equals compares two RedelegationEntry objects
+// Note the UnbondingId are not compared
```

Comment:
I am not sure about that, before we were using `String()` to compare, but `String()` was not including the UnbondingID. I am curious if the `.String()` implementation was correct, and should be replicated here. Or it was wrong, and then we should compare UnboardingId here. cc @amaurym

Reply:
- Is this equality checked somewhere in the non-test code?
- I feel like comparing all fields is the right thing to do.

Reply:
I agree, all field should be checked for verifying equality.
It seems like it is actually something missed in this PR: https://github.com/cosmos/cosmos-sdk/pull/13122 when the fields have been added.

Reply:
I think we should update the `.String()` function in 0.47 then or backport this. Wdyt?

Reply:
Yeah, we can update and backport the `String()` function, that's probably a correct thing to do.

But I don't think we should add the `.Equals` method here. We can use `proto.Equal(m1, m2)` in the test.

Reply:
I didn't know about it. That works!